### PR TITLE
Identity-vetting implementation

### PR DESCRIPTION
**Pull request by MKodde**
See point 3 of https://www.pivotaltracker.com/story/show/184191063 for details

### CODE REVIEW

File: src/Surfnet/StepupGateway/GatewayBundle/Tests/Entity/SecondFactorTest.php [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+            'self-asserted tiqr satisfies loa1.5' => ['tiqr', 1.5, true, false],
+            'self-asserted sms satisfies loa1.5' => ['sms', 1.5, true, false],
+        ];
+    }
+
+    public function provideErroneousIdentityVettedTestData()
+    {
+        return [
+            'unknown token type' => ['tjoepiekey', 3.0, "The Loa level of this type: tjoepiekey can't be retrieved."],
+            'unknown loa level' => ['yubikey', 0, 'Unknown loa level "0", known levels: "1", "1.5", "2", "3"'],
```
Comment (Shaky212) [inline src/Surfnet/StepupGateway/GatewayBundle/Tests/Entity/SecondFactorTest.php:141]: An idea to add some more edge cases like 1.4, 1.6, 4?

**Review by Shaky212 — APPROVED** [review]
Nice test work! Maybe you can add some LoA edge cases to test the lower and upper bindings.

File: src/Surfnet/StepupGateway/GatewayBundle/Tests/Entity/SecondFactorTest.php (hunk 1 above)
Comment (tvdijen) [reply src/Surfnet/StepupGateway/GatewayBundle/Tests/Entity/SecondFactorTest.php:141]: All I do is deliver edge-cases to you guys ;) I'm all in favor

Comment (MKodde) [reply src/Surfnet/StepupGateway/GatewayBundle/Tests/Entity/SecondFactorTest.php:141]: Haha, I attempted to test the possible exception avenues. But I can spice them up a notch :)

Comment (MKodde) [reply src/Surfnet/StepupGateway/GatewayBundle/Tests/Entity/SecondFactorTest.php]: I added some tests, and lo and behold. I found a bug in the Stepup-Bundle. The exception message casts the loA level to an integer value. Resulting in a message like (when specifying a level 3.5 token): 

'Invalid LoA level 3, valid levels are 1, 2, 3, 1.5'

Given this is a bug in another component, I'm not fixing this here. But instead add a report of this issue on the stepup backlog.
